fix: Mobile Responsiveness

nhcarrigan opened the pull request:
This PR will address the mobile responsiveness, in relation to #15. 

Fixes:
- [X] Title text too large
- [X] Title overlaps with header (I fixed this with a hard-coded margin, which doesn't seem to impact the desktop display)
- [X] Repository card too wide
- [X] Topic tags styled to reflect GitHub

Repository card is fixed by removing the `flex` on smaller screens and allowing the buttons to size inward, and hiding the information (e.g. language, PRs, forks) on smaller screens.

lukeocodes commented:
@netlify redeploy

lukeocodes commented:
Can you rebase, also I was considering removing the Mobile part of the menu because the Desktop menu is not complicated enough to not just use responsive classes on the menu items itself (and greatly simplifies that component).

Love it so far though, really good. Thanks!

nhcarrigan commented:
I dropped the mobile menu, and tweaked the desktop menu + announcement banner to play nicer with mobile/smaller screens. 

Also, squashed the commits because (as is usual when I `git anything`) I exploded my rebase attempt.
